tests/h/views/admin_organizations_test.py

```diff
@@ -0,0 +1,175 @@
+        def call_on_success(request, form, on_success, on_failure):
+            return on_success({'name': 'New org',
+                               'authority': 'example.org',
+                               'logo': '<svg>a logo</svg>'})
+        handle_form_submission.side_effect = call_on_success
+        ctrl = OrganizationCreateController(pyramid_request)
+
+        ctrl.post()
+
+        org = (pyramid_request.db.query(Organization)
+                              .filter_by(name='New org')
+                              .one())
+        assert org.authority == 'example.org'
+        assert org.logo == '<svg>a logo</svg>'
+
+    def test_post_redirects_to_list_view(self, pyramid_request, handle_form_submission):
+        def call_on_success(request, form, on_success, on_failure):
+            return on_success({'name': 'New org',
+                               'authority': 'example.org',
+                               'logo': '<svg>a logo</svg>'})
+        handle_form_submission.side_effect = call_on_success
+        ctrl = OrganizationCreateController(pyramid_request)
+
+        response = ctrl.post()
+
+        list_url = pyramid_request.route_url('admin_organizations')
+        assert response == redirect_302_to(list_url)
+
+
+@pytest.mark.usefixtures('routes')
+class TestOrganizationEditController(object):
+    def test_read_presents_org(self, pyramid_request, org):
+        ctrl = OrganizationEditController(org, pyramid_request)
+        ctx = ctrl.read()
+        assert ctx['form'] == self._expected_form(org)
+
+    def test_logo_is_empty_if_not_set(self, pyramid_request, org):
+        org.logo = None
+        ctrl = OrganizationEditController(org, pyramid_request)
+
+        ctx = ctrl.read()
+
+        assert ctx['form']['logo'] == ''
+
+    def test_read_shows_delete_button(self, pyramid_request, org):
+        ctrl = OrganizationEditController(org, pyramid_request)
+        ctx = ctrl.read()
+        assert ctx['delete_url'] == pyramid_request.route_url('admin_organizations_delete', pubid=org.pubid)
+
+    def test_read_does_not_show_delete_button_for_default_org(self, pyramid_request, org):
+        org.pubid = '__default__'
+        ctrl = OrganizationEditController(org, pyramid_request)
+
+        ctx = ctrl.read()
+
+        assert ctx['delete_url'] is None
+
+    def test_update_saves_org(self, pyramid_request, org, handle_form_submission):
+        def call_on_success(request, form, on_success, on_failure):
+            return on_success({'name': 'Updated name',
+                               'authority': org.authority,
+                               'logo': '<svg>new logo</svg>'})
+        handle_form_submission.side_effect = call_on_success
+        ctrl = OrganizationEditController(org, pyramid_request)
+
+        ctx = ctrl.update()
+
+        assert org.name == 'Updated name'
+        assert org.logo == '<svg>new logo</svg>'
+        assert ctx['form'] == self._expected_form(org)
```

Comment:
Of course! These test classes are just classes after all; wrapping up stuff in methods is a completely reasonable way to do these sorts of things (it had never occurred to me before!).